uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1002.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1002 {
+
+  /**
+   * Creates the 'other' sickness reason option value.
+   *
+   * @return bool
+   */
+  public function upgrade_1002() {
+    try {
+      $result = civicrm_api3('OptionValue', 'get', [
+        'sequential' => 1,
+        'option_group_id' => 'hrleaveandabsences_sickness_reason',
+        'name' => 'other',
+      ]);
+
+      if ($result['count'] == 0) {
+        civicrm_api3('OptionValue', 'create', [
+          'option_group_id' => 'hrleaveandabsences_sickness_reason',
+          'name' => 'other',
+          'label' => 'Other - Please leave a comment',
+          'value' => 12,
+          'weight' => 11,
+          'is_reserved' => 1,
+          'is_default' => 0,
+          'is_active' => 1
+        ]);
+      }
+    } catch(Exception $e) {
+      // We run all the upgraders during the extension installation, but, during
```

Comment:
I'm not sure I agree with this logic. 

It seems like if step 1002 depends on an option group then it should have been installed either in the install hook or in one of the earlier upgraders. 

It's also a "catch all", so if there is some legitimate problem here, like one of the parameters is changed and it breaks the creation call, it will be ignored because of the try-catch.

What advantage do you get from adding it both to the xml and here?